homeassistant/components/backup/manager.py

```diff
@@ -45,6 +47,23 @@ def __init__(self, hass: HomeAssistant) -> None:
         self.backing_up = False
         self.backups: dict[str, Backup] = {}
         self.loaded = False
+        self._backup_finish_callbacks: list[Awaitable] = []
+        self._backup_start_callbacks: list[Awaitable] = []
+
+    def register_backup_callback(
+        self,
+        *,
+        finish: Awaitable | None = None,
+        start: Awaitable | None = None,
```

Comment:
Why allow either to be `None`.

If you want to act based on that a backup is in progress, you need to make sure you stop doing something when a backup starts and resume that when backups stops.

Reply:
Right, but if we need to stop certain things from happening, we should block them from happening until we're done. I can't see how we would have only 1 of the 2.

Reply:
One case of start only I can see being implemented, is to flush registries and reproducible states to .storage

Reply:
That's fair